controller/openid_configuration.go

```diff
@@ -0,0 +1,86 @@
+package controller
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/fabric8-services/fabric8-auth/app"
+	"github.com/fabric8-services/fabric8-auth/client"
+	"github.com/fabric8-services/fabric8-auth/errors"
+	"github.com/fabric8-services/fabric8-auth/jsonapi"
+	"github.com/fabric8-services/fabric8-auth/log"
+	"github.com/fabric8-services/fabric8-auth/rest"
+	"github.com/goadesign/goa"
+)
+
+// OpenidConfigurationController implements the openid_configuration resource.
+type OpenidConfigurationController struct {
+	*goa.Controller
+	configuration LoginConfiguration
+}
+
+// NewOpenidConfigurationController creates a openid_configuration controller.
+func NewOpenidConfigurationController(service *goa.Service, configuration LoginConfiguration) *OpenidConfigurationController {
+	return &OpenidConfigurationController{Controller: service.NewController("OpenidConfigurationController"), configuration: configuration}
+}
+
+// Show runs the show action.
+func (c *OpenidConfigurationController) Show(ctx *app.ShowOpenidConfigurationContext) error {
+
+	// Getting Keycloak's /.well-known/openid-configuration
+	keycloakOpenIDConfigurationEndpoint := c.configuration.GetKeycloakURL() + "/auth/realms/" + c.configuration.GetKeycloakRealm() + "/.well-known/openid-configuration"
+	response, err := http.Get(keycloakOpenIDConfigurationEndpoint)
```

Comment:
Why loading it from Keycloak? We should provide provide this configuration independently from Keycloak.

Reply:
The idea was to show that for these properties we are directly using keycloak and whenever we decide to change that on keycloak, we wont need to change our openid-configuration. But if you would prefer not loading from keycloak, that would certainly make things simpler.

Reply:
https://github.com/fabric8-services/fabric8-auth/pull/277/commits/2af8a55888dcbefdd6b9a45c4064855c9c56ff32